include/mbedtls/aes.h

```diff
@@ -295,6 +295,46 @@ int mbedtls_aes_crypt_cfb8( mbedtls_aes_context *ctx,
                     unsigned char *output );
 #endif /*MBEDTLS_CIPHER_MODE_CFB */
 
+#if defined(MBEDTLS_CIPHER_MODE_OFB)
+/**
+ * \brief This function performs an AES-OFB (Output Feedback Mode) encryption
+ *        or decryption operation.
+ *
+ *        For OFB, you must set up the context with mbedtls_aes_setkey_enc(),
+ *        regardless of whether you are performing an encryption or decryption
+ *        operation. This is because OFB mode uses the same key schedule for
+ *        encryption and decryption.
+ *
+ *        The OFB operation is identical for encryption or decryption, therefore
+ *        no operation mode needs to be specified.
+ *
+ * \note  Upon exit, the content of the IV is updated so that you can
+ *        call the same function again on the next
+ *        block(s) of data and get the same result as if it was
+ *        encrypted in one call. This allows a "streaming" usage.
+ *        If you need to retain the contents of the
+ *        IV, you must either save it manually or use the cipher
+ *        module instead.
+ *
+ *
```

Comment:
Please add an explicit warning that IV must absolutely be unique for each message encrypted with the same key, or confidentiality is lost for the messages encrypted with the same IV and key. Please also mention that due to the nature of OFB, new IVs must NOT be generated by applying AES to a previous IV. On the other hand, IVs need not be unpredictable, so using a message counter for them is fine. (This is essentially a condensed version of the parts of Appendix C of SP800-38a that apply to OFB.)

Reply:
> Please add an explicit warning that IV must absolutely be unique for each message encrypted with the same key, or confidentiality is lost for the messages encrypted with the same IV and key.

Done.

> Please also mention that due to the nature of OFB, new IVs must NOT be generated by applying AES to a previous IV.

Are you sure we should state that? It's a negative, and it's unusual to state in documentation what people *shouldn't* do.

> On the other hand, IVs need not be unpredictable, so using a message counter for them is fine.

The security requirements of IV's are different for each block mode. If we say too much for OFB, I'm worried that people also think it might apply to other block modes, unless there's also some equivalent statement there.

For instance, it's ok for OFB to be predictable and use a message counter, but not for CBC or CFB. I think the documentation needs to be addressed there as well.

Reply:
> Are you sure we should state that? It's a negative, and it's unusual to state in documentation what people shouldn't do.
I'm not sure. That negative recommendation is in the NIST document. but perhaps it doesn't belong in our documentation.

> The security requirements of IV's are different for each block mode. If we say too much for OFB, I'm worried that people also think it might apply to other block modes, unless there's also some equivalent statement there.
Then I think we should provide guidance for other functions as well. Since this is a slightly tricky subject that can have a major impact on security, I don't think we should leave users on their own.